### DISCUSSION TITLE
Manually rollback

### DISCUSSION THREAD

**Discussion by joaqquin89** [fluxcd/flux2#2916] [General] [answered]
Hi Guys 

I have an doubt , because I would like to do the rollbacks but manually (to integrate the feature with a  bot or that is the idea).

I saw this part in the documentation https://fluxcd.io/docs/guides/webhook-receivers/ but that endpoint could be works ? 

bests

**Comment by kingdonb** [comment, marked as answer]
Greetings and thanks for participating in the Flux community!

Rollbacks in Flux are generally performed in Git. Flux will deploy whatever is at the HEAD of the branch or tag ref of the GitRepository that you give it to synchronize, and filtered through the lens of the Kustomization. Flux does not manage rollbacks, except in the case of a commit with `sync.wait` enabled that failed due to health checking – but that's a bit different from a rollback.

Could you write more about what you're trying to accomplish here? It wasn't clear from your post. Webhook receivers in Flux will only trigger the normal action of Flux. The Flagger Canary resource has a `rollback webhook` which might also be what you're looking for:

https://docs.flagger.app/usage/webhooks

**Reply by dheerajkasaudhan** [reply]
Hi everyone

I just wanted to know that ,if there is any option for rollback in fluxCD.
Because If we didn't want to deploy the latest version and wanted to go back to previous deployment then what the option of roll back in fluxCD.. is there any flux CD command or anything else ..

**Reply by nagarjunacse02** [reply]
Even i have same query, can someone help on this pls

**Reply by VinGarcia** [reply]
@nagarjunacse02 and @dk2645 I am searching for how to do this, but I am pretty sure that if there is an existing solution supported by flux it would work in a similar manner to how the image update automation works: It would first detect something went wrong and then revert the commit that caused that issue and commit that to your branch.

Because the internal state of flux on the cluster should always reflect the branch or tag referenced by the GitRepository used as source.
